### ISSUE TITLE
frule invalidation broken

### ISSUE THREAD

**Issue by oxinabox** [JuliaDiff/Diffractor.jl#269]
We actually have a test for this: https://github.com/JuliaDiff/Diffractor.jl/blob/b37f585d4ba8fa5cb607cd75aaabb5464eef6633/test/rules.jl
but it turns out this file has never been linked up to the `test/runtests.jl` file.
So it wasn't getting run by CI.

The last test in that file fails.
Both on 1.10 and on nightly

**Comment by oxinabox** [comment]
We do manually insert an edge to the MethodInstance being AD'ed here
https://github.com/JuliaDiff/Diffractor.jl/blob/b37f585d4ba8fa5cb607cd75aaabb5464eef6633/src/codegen/forward.jl#L114

Do we need to manually insert an edge to every method we call in the generated code?
Surely not, we haven't run type-inference so we can't even know that, and that surely already will insert its own edges, right?

**Comment by Keno** [comment]
No, but we might need an edge for the frule of the MethodInstance we're currently ADing to cover the case where an frule is added to something that had none before.

**Comment by oxinabox** [comment]
I don't think so.
That case would get caught from the calling code, no?
Since `∂☆` always calls `∂☆internal` first
and that does  (basically):
```
function ∂☆internal(...)
    res = frule(...)
    if nothing===res
        ∂☆recurse...  # this code we are talking about
    else
        shuffle_base(res)
    end
end
```
So adding a new frule should already invalidate it, since it invalidates the top line of  `∂☆internal`

**Comment by Keno** [comment]
Ah yes, probably. Something else going on then.
